server.js

```diff
@@ -20,4 +20,57 @@ app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
+app.get('/messages', function (request, response) {
+  response.json(messages)
+})
+
+app.post('/messages', function (request, response) {
+  let newMessage = {};
+
+  if (request.body.from.trim() && request.body.text.trim()) {
+    newMessage.id = messages.length
+    newMessage.from = request.body.from
+    newMessage.text = request.body.text
+    newMessage.timeSent = new Date()
+    messages.push(newMessage)
+    response.json(messages)
+  } else {
+    response.status(400).json("Please fill all fileds")
+  }
+
+})
+
+app.get('/messages/search', function (request, response) {
+  let searchWord = request.query.text
+  let result = messages.filter(item => item.text.toLowerCase().includes(searchWord.toLowerCase()))
+  if (result.length > 0) {
+    response.json(result)
+  } else {
+    response.status(404).json("Nothing was found")
+  }
+})
+
+app.get('/messages/latest', function (request, response) {
+  let latestMessages = messages.splice(-2)
+  response.json(latestMessages)
+})
+
+app.get('/messages/:id', function (request, response) {
+  let id = parseInt(request.params.id)
+  let result = messages.find(item => item.id === id)
+  if (result !== undefined) {
+    response.json(result)
+  } else {
+    response.json("not found")
```

Comment:
The logic for the error handling works well. This is where a 404 status code should be used - when a resource is requested via an ID or name in the path parameter and this does not exist.

Reply:
```suggestion
  response.status(404).json("not found")
```
This line should work